### DISCUSSION TITLE
Converting a PySCIPOpt model to Ecole doesn't seem to work

### DISCUSSION THREAD

**Discussion by zohairshafi** [ds4dm/ecole#357] [Questions] [answered]
I've been trying to solve a simple set cover problem using Ecole with the intention of scaling it up and solving it using the GCNN based branching mechanism (https://github.com/ds4dm/ecole/blob/master/examples/branching-imitation/example.ipynb)

Here is the definition in PySCIPOpt

```# Define the set of elements and the set of subsets
elements = set(range(1, 6))
subsets = {
    1: set([1, 2, 3]),
    2: set([2, 4]),
    3: set([3, 5]),
    4: set([4, 5])
}

model = Model("set_cover")

# Create a binary variable for each subset
subset_vars = {}
for i in subsets:
    subset_vars[i] = model.addVar(vtype="I", name=f"subset_{i}")

# Add the set cover constraint
for j in elements:
    model.addCons(sum(subset_vars[i] for i in subsets if j in subsets[i]) >= 1)

# Set the objective to minimize the number of selected subsets
model.setObjective(sum(subset_vars[i] for i in subsets), "minimize")
```

Calling `model.optimize()` here works perfectly fine and I get a solution. 

However, when I try to wrap it using Ecole, it does not work and the environment reset return done = True. 

```
scip_model = model
ecole_model = ecole.scip.Model.from_pyscipopt(scip_model)
scip_parameters = {
    "separating/maxrounds": 0,
    "presolving/maxrestarts": 0,
    "limits/time": 3600,
}
env = ecole.environment.Branching(
    observation_function=(
        ExploreThenStrongBranch(expert_probability=0.05),
        ecole.observation.NodeBipartite(),
    ),
    information_function={
        "nb_nodes": ecole.reward.NNodes(),
        "time": ecole.reward.SolvingTime(),
    },
    scip_params=scip_parameters,
)

env.reset(ecole_model) 
# returns (None, None, 1.0, True, {'nb_nodes': 0.0, 'time': 0.001369443})
```

Solving it using the steps defined in the branching imitation notebook returns the following : 
```
Instance   0 | SCIP nb nodes       0  | SCIP time     0.00 
                      | GNN  nb nodes       0  | GNN  time     0.00
 ```

Is there something I'm doing wrong? 

(This works perfectly well with the inbuilt set cover instance generator though, but I would like to pass in my own constraints)

Thank you!

**Comment by kevin031060** [comment]
Have you solved this problem? I meet the same issue

**Reply by zohairshafi** [reply, marked as answer]
If I remember right, the way it expects the objective was a little different. Here is some code I used that might work (sorry I've moved away from working on this so don't exactly remember if this is how I worked around it) : 

    def scip_solve(universe, subsets, costs):

         s = time.time()
         # Create Model
         model = pyscipopt.Model("Set Cover")

         # create a binary variable for each subset
         vars = []
         for i in range(len(subsets)):
                  var = model.addVar("x_{}".format(i), vtype="B")
                  vars.append(var)

         # set the objective function to minimize the total cost of the selected subsets
         model.setObjective(sum(costs[i] * vars[i] for i in range(len(subsets))), "minimize")

         # add a constraint for each element in the ground set
         # the element must be covered by at least one of the selected subsets
         for e in universe:
                  model.addCons(sum(vars[i] for i in range(len(subsets)) if e in subsets[i]) >= 1)

         return model

```
inst = {}
inst['universe'] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14}
inst['subsets'] = {
                   0 : {1, 2}, 
                   1 : {3, 4, 5, 6}, 
                   2 : {7, 8, 9, 10, 11, 12, 13, 14}, 
                   4 : {1, 3, 4, 7, 8, 9, 10}, 
                   5 : {2, 5, 6, 11, 12, 13, 14},
                  }
inst['solutions'] = [['4', '5']]
inst['costs'] = [1, 1, 1, 1, 1]

model = scip_solve(universe = inst['universe'], inst['subsets'], inst['costs'])

```
The model returned should be compatible with the Ecole environment.

**Reply by kevin031060** [reply]
Thanks a lot! It works!
